src/components/PeoplePage.tsx

```diff
@@ -1,8 +1,110 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    (async () => {
+      try {
+        const response = await getPeople();
+
+        setPeople(response);
+        setIsLoading(false);
+      } catch {
+        setError(true);
+        setIsLoading(false);
+      }
+    })();
+  }, []);
+
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const filteredPeople = people
+    .filter(person => {
+      if (sex) {
+        return person.sex === sex;
+      }
+
+      return true;
+    })
+    .filter(person => {
+      if (query) {
+        return (
+          person.name.toLowerCase().includes(query.toLowerCase()) ||
+          person.motherName?.toLowerCase().includes(query.toLowerCase()) ||
+          person.fatherName?.toLowerCase().includes(query.toLowerCase())
+        );
+      }
+
+      return true;
+    })
+    .filter(person => {
+      if (centuries.length > 0) {
+        const century = Math.ceil(person.born / 100);
+
+        return centuries.includes(String(century));
+      }
+
+      return true;
+    });
+
+  if (sort) {
+    switch (sort) {
+      case 'name':
+        const result = filteredPeople.sort((person1, person2) => {
+          return person1.name
+            .toLowerCase()
+            .localeCompare(person2.name.toLowerCase());
+        });
+
+        if (order === 'desc') {
+          result.reverse();
+        }
+
+        break;
+      case 'sex':
+        const result2 = filteredPeople.sort((person1, person2) => {
+          return person1.sex
+            .toLowerCase()
+            .localeCompare(person2.sex.toLowerCase());
+        });
+
+        if (order === 'desc') {
+          result2.reverse();
+        }
+
+        break;
+
+      case 'born':
+      case 'died':
+        const result3 = filteredPeople.sort((person1, person2) => {
+          return person1.born - person2.born;
```

Comment:
The sorting logic for 'born' and 'died' cases is currently identical, both sorting by 'born'. If you intend to sort by 'died' as well, you should adjust the logic to compare 'person1.died - person2.died' for the 'died' case.